src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java

```diff
@@ -113,25 +114,30 @@
     }
 
     default ThrowableProblem toProblem(final Throwable throwable) {
-        final StatusType status = resolveStatus(throwable)
-                .<StatusType>map(HttpStatusAdapter::new)
+        final StatusType status = Optional.ofNullable(resolveResponseStatus(throwable))
+                .map(responseStatusToStatusType())
                 .orElse(Status.INTERNAL_SERVER_ERROR);
 
         return toProblem(throwable, status);
     }
 
-    default Optional<HttpStatus> resolveStatus(final Throwable type) {
-        @Nullable final ResponseStatus candidate = findMergedAnnotation(type.getClass(), ResponseStatus.class);
-
-        if (candidate == null) {
-            if (type.getCause() == null) {
-                return Optional.empty();
+    default Function<ResponseStatus, StatusType> responseStatusToStatusType() {
```

Comment:
Should be 

```java
default StatusType toStatusType(final ResponseStatus status) {
```

and use it as `this::toStatusType`. 

Do we actually need to have this method overridable? Could it be static inside the pojo class directly?

Reply:
I thought it was interesting to have it configurable. But of course it can be done as a static class method.

Reply:
I'd like to keep the public API as minimal as possible. It's easy to add it later,but it's hard to remove.